rustler/src/types/binary.rs

```diff
@@ -1,43 +1,105 @@
+//! Safe wrappers around Erlang binaries.
+//!
+//! Rustler provides two types for wrapping Erlang binaries: [`Binary`] and
+//! [`OwnedBinary`].  Both represent a contiguous region `u8`s, and they both use
+//! the Erlang allocator. The primary difference between the two is ownership
+//! semantics.
+//!
+//! The _owned_ in `OwnedBinary` refers to the fact that it owns the VM binary is
+//! wraps, and the _owner_ of an `OwnedBinary` is free to modify its contents
+//! without fear of access violations or upsetting the VM. Ownership lasts until it
+//! is dropped by either falling out of scope, or converting it into a regular
+//! `Binary` via [`to_owned`] or one of the conversion traits. Once dropped, the VM
+//! is allowed to the free the backing memory at its leisure. Similar to `Vec`,
+//! `OwnedBinary` does not implement `copy` [^1], so passing one around by value is
```

Comment:
Nit: In Rust, if something implements `Copy`, the assumption is usually that it can by efficiently passed by value. Since `Vec`s can be `clone`d, I guess it would be better to leave out that comparison and just say

     An `OwnedBinary` can not be copied or cloned and is thus always moved.

Reply:
>  the assumption is usually that it can by efficiently passed by value

Yeah. I was trying to go into a little more detail than you'd normally see in rust documentation. Except for me, everyone else at my company using Rustler is an Erlang-first developer, with varying familiarity with Rust. I think that might be case for other users of this crate. 

I guess what I'm really trying to say there is "don't worry, it's cheap to move these binary types around", but I think your suggestion makes the most sense.